core/services/shell_commands.rb

```diff
@@ -26,57 +28,102 @@ def self.environment
     @env
   end
 
+  # Log stdout and stderr
+  #
+  # @param logger [Out] logger to log information to
+  # @param stdout [IO] stdout stream
+  # @param stderr [IO] stderr stream
+  # @param stdout_text [String] string to save history of stdout stream
+  # @param stderr_text [String] string to save history of stderr stream
+  # @return [Array<IO>, String, String] array of streams (or nil if stream has ended), stdout history, stderr history.
+  def self.log_command_streams(logger, stdout, stderr, stdout_text, stderr_text)
+    wait_streams = []
+    wait_streams << read_stream(stdout) do |line|
+      logger.info(line)
+      stdout_text += line
+    end
+    wait_streams << read_stream(stderr) do |line|
+      logger.error(line)
+      stderr_text += line
+    end
+    [wait_streams.compact, stdout_text, stderr_text]
+  end
+
+  # Handle the command inactivity, log messages about it, kill command process if break_on_inactivity is true
+  #
+  # @param logger [Out] logger to log information to
+  # @param show_notifications [Boolean] show notifications when there is no
+  # @param command_inactivity_timeout [Number] the timeout after which the command is considered inactive
+  # @param command [String] running command
+  # @param command_was_inactive [Boolean] flag stating that the command is re-considered (timeout * 2) inactive
+  # @param pid [Number] pid of inactivity command
+  # @param break_on_inactivity [Boolean] flag to terminate command execution when it is idle
+  # @return [Boolean] true if command process was killed, otherwise - false
+  # rubocop:disable Metrics/ParameterLists
+  def self.handle_inactivity(logger, show_notifications, command_inactivity_timeout, command,
```

Comment:
The method name is too vague, it does not describe the reason for method to exist.